srv/src/main/java/com/sap/db/util/SocketUtils.java

```diff
@@ -0,0 +1,22 @@
+package com.sap.db.util;
+
+import java.net.Socket;
+import java.nio.channels.AsynchronousSocketChannel;
+
+// Override HANA class, which uses feature not supported in Native Images
+// See https://github.com/oracle/graal/issues/6457
+public final class SocketUtils {
+	private SocketUtils() {
+		throw new AssertionError("Non-instantiable class");
+	}
+
+	public static void setKeepAliveOptions(Socket socket,
+			int tcpKeepAliveIdle, int tcpKeepAliveInterval, int tcpKeepAliveCount) {
+
+	}
+
+	public static void setKeepAliveOptions(AsynchronousSocketChannel channel,
+			int tcpKeepAliveIdle, int tcpKeepAliveInterval, int tcpKeepAliveCount) {
+
+	}
+}
```

Comment:
haha.. ok. Let's not forget that one later. 😎🤷🏾‍♂️

Reply:
Yes, but this class is anyway only present in JDK 17 (since the features it uses are also JDK17 only). On JDK < 17 the HANA driver itself brings an empty implementation. So it is "officially supported" to deactivate this code :)
But I'll hope that the GraalVM bug is fixed and we can remove this again completely.